File: server/models/Transaction.py
// models/Transaction.js

const { DataTypes } = require("sequelize");
const sequelize = require("../config/database");
const Account = require("./Account");

const Transaction = sequelize.define("Transaction", {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  accountId: {
    type: DataTypes.INTEGER,
    references: {
      model: Account,
      key: "id",
    },
    allowNull: false,
  },
  type: {
    type: DataTypes.ENUM("credit", "debit"),
    allowNull: false,
  },
  amount: {
    type: DataTypes.DOUBLE,
    allowNull: false,
    validate: {
      min: 0,
    },
  },
  date: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
    allowNull: false,
  },
  description: {
    type: DataTypes.STRING,
  },
});

module.exports = Transaction;

File: server/models/Transaction.js
const { pool } = require('../config/database');

const createTransactionTable = async () => {
  const query = `
    CREATE TABLE IF NOT EXISTS transactions (
      id SERIAL PRIMARY KEY,
      accountId INTEGER REFERENCES accounts(id),
      type VARCHAR NOT NULL,
      amount DOUBLE PRECISION NOT NULL,
      date TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
      description VARCHAR,
      journalType VARCHAR
    );
  `;
  await pool.query(query);
};

createTransactionTable().catch(err => console.error('Error creating transaction table:', err));

module.exports = { createTransactionTable };
